### PR TITLE
Post explore update

### PR DESCRIPTION
**Pull request by robvk**

### CODE REVIEW

File: Week4/README.md [hunk 1]
```diff
@@ -4,27 +4,22 @@
...
-   - Array methods
-2. [Closures](https://study.hackyourfuture.net/#/javascript/closures)
-   - Execution context
-   - Why do we need closures?
-3. [Problem solving](https://study.hackyourfuture.net/#/programming/problem-solving-process)
-4. [Unit testing](https://study.hackyourfuture.net/#/testing/unit-tests)
-   - [Testing framework (Jest)](https://study.hackyourfuture.net/#/tools/test-framework)
-5. [Finalise Your personal brand](https://github.com/HackYourFuture/yourpersonalbrand)
+1. [The `this` keyword](https://study.hackyourfuture.net/#/javascript/this)
+1. [Constructor functions](https://study.hackyourfuture.net/#/javascript/constructor-functions)
```
Comment (remarcmij) [inline Week4/README.md:8]: For the 'Miss Xing' video in the study book you may want to suggest students to turn on subtitles in YouTube if they have difficulty understanding her foreign accent.

File: Week4/prep-exercises/1-wallet/README.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
+# Wallet Exercise
```
Comment (remarcmij) [inline Week4/prep-exercises/1-wallet/README.md:1]: When writing this exercise I had assumed that students would have completed the week4 ex5-wallet exercise on which the current exercise builds. As the homework for week 4 will be dropped I think there is now missing context here. Also, ex5-wallet teaches how to use the Chrome debugger wich students will now miss out on. Perhaps we can move ex5-wallet from the homework repo to week 4 here as a prep-exercise and intro into this exercise?

File: Week4/prep-exercises/1-wallet/README.md [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+
+4. Update the other methods as required to support the new functionality.
+
+## Some help
+
+As this will not be easy, let's go through a few of the concepts again but then specifically to this exercise:
+
+### Public Interface vs Private Implementation
+
+Imagine that our various wallet implementations (however simplistic they may be) will be used as libraries by third party developers, perhaps as part of a banking application. We would expect those developers to access the functionality of our wallet through the methods that we provide for that purpose. We consider that to be the _public interface_ of our wallet, i.e. a contract between ourselves as library owner and the users of the library. If third party developers would be allowed to manipulate the internals of the wallet (e.g. `cash`, `dailyAllowance`, etc.) we can no longer guarantee its correct functioning. We consider those internals to be _private implementation details_, and we would want the reserve the right to make implementation changes/improvements. So long as such changes have no impact on what can be observed through the _public interface_ (i.e. we honour the "contract") there should not be a problem.
```
Comment (remarcmij) [inline Week4/prep-exercises/1-wallet/README.md:34]:
```suggestion
Imagine that our various wallet implementations (however simplistic they may be) will be used as libraries by third party developers, perhaps as part of a banking application. We would expect those developers to access the functionality of our wallet through the methods that we specifically provide for that purpose. We consider that to be the _public interface_ of our wallet, i.e. a contract between ourselves as library owner and the users of the library. If third party developers would be allowed to manipulate the internals of the wallet (e.g. `cash`, `dailyAllowance`, etc.) we can no longer guarantee its correct functioning. We consider those internals to be _private implementation details_, and we would want the reserve the right to make implementation changes/improvements. So long as such changes have no impact on what can be observed through the _public interface_ (i.e. we honour the "contract") there should not be a problem.
```

File: Week4/overview.ipynb [hunk 4]
```diff
@@ -0,0 +1,463 @@
...
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Objects, Classes, new, this and prototypes\n",
+    "\n",
+    "> Created by the one and only Jim, you can find him on our [Slack](https://hackyourfuture.slack.com/team/U383PTTK9) and on [GitHub](https://github.com/remarcmij)!\n",
+    "\n",
+    "> This is a Jupyter notebook, you can run the code immediately if you do some setting up steps, although it is not necessary to do so as we have provided the results. If you want to set it up you can go through the following [article](https://www.tomche.space/post/using-javascript-kernel-in-vscode-jupyter-notebooks/)\n",
```
Comment (remarcmij) [inline Week4/overview.ipynb:12]: Not sure if it is worth the trouble for students to install all the prerequisites for using Notebooks with JS. They will not need it for the rest of the curriculum and they may waste time and energy getting this working. I would suggest to treat this as read-only for them.

**Review by remarcmij — COMMENTED** [review]
Hi Rob, happy to approve at any instant you need it, but for now have some suggestions for you to consider.

File: Week4/prep-exercises/1-wallet/README.md (hunk 2 above)
Comment (robvk) [reply Week4/prep-exercises/1-wallet/README.md:1]: That is actually still the case, the wallet exercise is now in week 3 so they will have all this information. We are not losing anything from the other weeks, it is just that students have already done the JavaScript basics during the application process so week 1 and 2 are now split differently and we can attach this week to the end

File: Week4/overview.ipynb (hunk 4 above)
Comment (robvk) [reply Week4/overview.ipynb:12]: Good point, will reword it to tell them not to do it as I think vscode might try to get them to once it identifies ipynb file name

**Comment by robvk** [comment]
Thanks for the comments and suggestions, have gone through them and adjusted!

**Review by remarcmij — APPROVED** [review]
Thanks for the updates, Rob, and for clarifying the situation around the homework I wrongly thought was dropped off.
